filemanager/azureblobmanager.go

```diff
@@ -0,0 +1,368 @@
+	containerURL, err := manager.getContainerURL()
+	if err != nil {
+		return UploadedFile{}, err
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, manager.getTimeout())
+	defer cancel()
+
+	if manager.createContainer() {
+		_, err = containerURL.Create(ctx, azblob.Metadata{}, azblob.PublicAccessNone)
+		err = manager.suppressMinorErrors(err)
+		if err != nil {
+			return UploadedFile{}, err
+		}
+	}
+
+	fileName := path.Join(manager.config.Prefix, path.Join(prefixes...), path.Base(file.Name()))
+
+	// Here's how to upload a blob.
+	blobURL := containerURL.NewBlockBlobURL(fileName)
+	_, err = azblob.UploadFileToBlockBlob(ctx, file, blobURL, azblob.UploadToBlockBlobOptions{
+		BlockSize:   4 * 1024 * 1024,
+		Parallelism: 16,
+	})
+	if err != nil {
+		return UploadedFile{}, err
+	}
+
+	return UploadedFile{Location: manager.blobLocation(&blobURL), ObjectName: fileName}, nil
+}
+
+func (manager *azureBlobManager) Download(ctx context.Context, output *os.File, key string) error {
+	containerURL, err := manager.getContainerURL()
+	if err != nil {
+		return err
+	}
+
+	blobURL := containerURL.NewBlockBlobURL(key)
+
+	ctx, cancel := context.WithTimeout(ctx, manager.getTimeout())
+	defer cancel()
+
+	// Here's how to download the blob
+	downloadResponse, err := blobURL.Download(ctx, 0, azblob.CountToEnd, azblob.BlobAccessConditions{}, false, azblob.ClientProvidedKeyOptions{})
+	if err != nil {
+		return err
+	}
+
+	// NOTE: automatically retries are performed if the connection fails
+	bodyStream := downloadResponse.Body(azblob.RetryReaderOptions{MaxRetryRequests: 20})
+
+	// read the body into a buffer
+	downloadedData := bytes.Buffer{}
+	_, err = downloadedData.ReadFrom(bodyStream)
+	if err != nil {
+		return err
+	}
+
+	_, err = output.Write(downloadedData.Bytes())
+	return err
+}
+
+func (manager *azureBlobManager) Delete(ctx context.Context, keys []string) (err error) {
+	containerURL, err := manager.getContainerURL()
+	if err != nil {
+		return err
+	}
+
+	for _, key := range keys {
+		blobURL := containerURL.NewBlockBlobURL(key)
+
+		_ctx, cancel := context.WithTimeout(ctx, manager.getTimeout())
+		_, err := blobURL.Delete(_ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
+		if err != nil {
+			cancel()
+			return err
+		}
+		cancel()
+	}
+	return
+}
+
+func (manager *azureBlobManager) Prefix() string {
+	return manager.config.Prefix
+}
+
+func (manager *azureBlobManager) GetObjectNameFromLocation(location string) (string, error) {
+	strToken := strings.Split(location, fmt.Sprintf("%s/", manager.config.Container))
+	return strToken[len(strToken)-1], nil
+}
+
+func (manager *azureBlobManager) GetDownloadKeyFromFileLocation(location string) string {
+	str := strings.Split(location, fmt.Sprintf("%s/", manager.config.Container))
+	return str[len(str)-1]
+}
+
+func (manager *azureBlobManager) suppressMinorErrors(err error) error {
+	if err != nil {
```

Comment:
[optional] instead of nesting we could invert the condition and remove the below code block from the `if`.